trpc/queries/goalV2.ts

```diff
@@ -0,0 +1,210 @@
+            qb.where('Goal.priorityId', 'in', params.query?.priority || []),
+        )
+        .$if(params.query?.stateType != null && params.query.stateType.length > 0, (qb) =>
+            qb.where('Goal.stateId', 'in', ({ selectFrom }) =>
+                selectFrom('State')
+                    .select('State.id')
+                    .where('State.type', 'in', params.query?.stateType || []),
+            ),
+        )
+        .$if(params.query?.state != null && params.query.state.length > 0, (qb) =>
+            qb.where('Goal.stateId', 'in', params.query?.state || []),
+        )
+        .$if(params.query?.tag != null && params.query.tag.length > 0, (qb) =>
+            qb.where('tag.id', 'in', params.query?.tag || []),
+        )
+        .$if(params.query?.participant != null && params.query.participant.length > 0, (qb) =>
+            qb.where('participant.activityId', 'in', params.query?.participant || []),
+        )
+        .$if(params.query?.query != null && params.query.query.length > 0, (qb) =>
+            qb.where(({ or, eb }) =>
+                or([
+                    eb('Goal.title', 'ilike', params.query?.query || ''),
+                    eb('Goal.description', 'ilike', params.query?.query || ''),
+                    eb('Goal.projectId', 'in', (qb) =>
+                        qb
+                            .selectFrom('Project')
+                            .select('Project.id')
+                            .where((eb) =>
+                                eb.or([
+                                    eb('Project.title', 'ilike', params.query?.query || ''),
+                                    eb('Project.description', 'ilike', params.query?.query || ''),
+                                ]),
+                            ),
+                    ),
+                ]),
+            ),
+        )
+        .$if(params.role === Role.USER, (qb) =>
+            qb.where('Goal.projectId', 'in', ({ selectFrom }) =>
+                selectFrom('Project')
+                    .select('Project.id')
+                    .where(({ eb, not, exists, selectFrom }) =>
+                        eb.or([
+                            eb('Project.activityId', '=', params.activityId),
+                            eb(
+                                'Project.id',
+                                'in',
+                                selectFrom('_projectAccess').select('B').where('A', '=', params.activityId),
+                            ),
+                            not(
+                                exists(
+                                    selectFrom('_projectAccess')
+                                        .select('B')
+                                        .where('_projectAccess.A', '=', params.activityId),
+                                ),
+                            ),
+                        ]),
+                    ),
+            ),
+        )
+        .limit(30)
```

Comment:
you probably should move it higher to a const or even an ENV

Reply:
This code now is useless, in projects' query module I created same variable